docs/docsite/rst/porting_guides/porting_guide_core_2.19.rst

```diff
@@ -766,6 +766,24 @@ Noteworthy plugin changes
   This filter now returns ``False`` instead of ``None`` when the input is ``None``.
   The aforementioned deprecation warning is also issued in this case.
 
+* Passing nested non-scalars with embedded templates that may resolve to ``Undefined`` to Jinja2
+  filter plugins such as ``default`` and ``mandatory``, and test plugins including ``defined`` and ``undefined``
+  no longer evaluate the same, since nested non-scalars with embedded templates are only templated on use.
```

Comment:
Perhaps invert things and start with the final clause. That clarifies the behaviour and the preceding two clauses clarify the results of that behaviour.